nft_ingester/src/program_transformers/bubblegum/cancel_redeem.rs

```diff
@@ -19,29 +21,43 @@ where
 {
     if let (Some(le), Some(cl)) = (&parsing_result.leaf_update, &parsing_result.tree_update) {
         let seq = save_changelog_event(cl, bundle.slot, txn).await?;
+        #[allow(unreachable_patterns)]
         return match le.schema {
             LeafSchema::V1 {
                 id,
-                delegate,
                 owner,
+                delegate,
                 ..
             } => {
-                let id_bytes = id.to_bytes().to_vec();
+                let id_bytes = id.to_bytes();
+                let owner_bytes = owner.to_bytes().to_vec();
                 let delegate = if owner == delegate {
                     None
                 } else {
                     Some(delegate.to_bytes().to_vec())
                 };
-                let owner_bytes = owner.to_bytes().to_vec();
-                let asset_to_update = asset::ActiveModel {
-                    id: Unchanged(id_bytes.clone()),
-                    leaf: Set(Some(le.leaf_hash.to_vec())),
-                    delegate: Set(delegate),
-                    owner: Set(Some(owner_bytes)),
-                    seq: Set(seq as i64), // gummyroll seq
-                    ..Default::default()
-                };
-                update_asset(txn, id_bytes, Some(seq), asset_to_update).await
+
+                // Partial update of asset table with just leaf.
+                upsert_asset_with_leaf_info(
```

Comment:
Have you factored in that these are now separate DB writes and on failure, it will only be partially complete?

Reply:
Yes I have thought about it seems like it is is OK, at least as OK as the current code with how it deals with DB write failures.

What I mean is that in the existing code, if (for example) a database write fails on `transfer`, then you just get a metric or log and the asset record for  `leaf`, `owner`, `delegate` and `seq` would be unchanged.  The `transfer` transaction would need to be rerun to fix it (which isn't 100% reliable given shared `seq`), or a later transaction that updates those fields.  So in the existing code when this happens the `asset` row is overall just partially correct.

With this PR's change, you could have a state where 'leaf` had been updated correctly but `owner`, `delegate` and `seq` may be out of date.  The `transfer` transaction or a later transaction would still need to be rerun to fix them, similar as before.  So again the overall state is one of partial correctness.  Just different partially correct fields than before.

Neither case auto-corrects for the failed write as far as I know and that would be a more extensive change to the backfilling process.

Reply:
That was also my take. Works for me.